flyteidl/protos/flyteidl/core/literals.proto

```diff
@@ -46,6 +46,29 @@ message Binary {
     string tag = 2;
 }
 
+// Represents a JSON object encoded as a byte array.
+// This field is used to store JSON-serialized data, which can include
+// dataclasses, dictionaries, Pydantic models, or other structures that
+// can be represented as JSON objects. When utilized, the data should be
+// deserialized into its corresponding structure.
+// This design ensures that the data is stored in a format that can be
+// fully reconstructed without loss of information.
+message Json {
+    // The JSON object serialized as a byte array.
+    bytes value = 1;
+
+    // The format used to serialize the byte array.
+    // This field identifies the specific format of the serialized JSON data,
+    // allowing future flexibility in supporting different JSON variants.
```

Comment:
add a comment here saying that serialization formats need to be supported in python, golang, rust, and javascript, in order for the full flyte experience to work.  (just throwing rust in there for now, we don't have anything yet).

Reply:
Yes, you're right, this is a pretty good and important advice, thank you.